kong.go

```diff
@@ -192,9 +192,13 @@ func (k *Kong) interpolateValue(value *Value, vars Vars) (err error) {
 	if value.Enum, err = interpolate(value.Enum, vars, nil); err != nil {
 		return fmt.Errorf("enum value for %s: %s", value.Summary(), err)
 	}
+	if value.Flag.Env, err = interpolate(value.Flag.Env, vars, nil); err != nil {
```

Comment:
Do we need to do a nil check on `value.Flag` here before proceeding? I didn't see any nil checks on `value` above, so I figured it would be available, but I haven't spent much time under the hood in this codebase.

Reply:
```go
if value.Flag != nil {
	// interpolate
}
```

Reply:
`TestHelpOptionalArgs` was failing without this nil check. Added it.